src/core/dbus-unit.c

```diff
@@ -782,7 +782,9 @@ static int bus_unit_method_freezer_generic(sd_bus_message *message, void *userda
         if (r == 0)
                 reply_no_delay = true;
 
-        assert(!u->pending_freezer_message);
+        if (u->pending_freezer_message) {
+                u->pending_freezer_message = sd_bus_message_unref(u->pending_freezer_message);
+        }
```

Comment:
this doesn't look right. this means the original client which asked for the freezing will just hang as it will never get a reply to its original message.

We should instead send back a reply with an explicit dbus error "FreezeCancelled" or so.

This requires some further changes, i.e. pending_freezer_message currently stores the prepared successful reply message already. We need to change that to instead contain the original method call message, so that we have the freedom to either return success or an error still. And then we return an error here before releasing the message.